src/dmd/audit_log.d

```diff
@@ -26,19 +26,23 @@ void Audit_Log(ASTNode node, string functioncall = __FUNCTION__, string moduleca
     {
         bool Exist()
         {
-            for(int x = 0; x < global.params.calls.length; x++)
+            for(size_t i = 0; i < global.params.calls.length; i++)
             {
-                if(global.params.calls[x].toString() == functioncall
-                   || global.params.calls[x].toString() ==  modulecall)
+                if(global.params.calls[i].toString() == functioncall
+                   || global.params.calls[i].toString() ==  modulecall)
                     return true;
             }
             return false;
         }
         if(!Exist())
             return;
-        auto x = "debug/" ~ modulecall ~ "/" ~ functioncall ~ ".txt\0";
-        auto y = fopen(x.ptr, "w+\0".ptr);
-        file[modulecall ~ " " ~ functioncall] = y;
+        auto tmpstring = "debug/" ~ modulecall ~ "/" ~ functioncall ~ ".txt\0";
+        auto tmpfile = file[modulecall ~ " " ~ functioncall];
+        if(tmpfile == null)
+        {
+            tmpfile = fopen(tmpstring.ptr, "w+\0".ptr);
+            file[modulecall ~ " " ~ functioncall] = tmpfile;
+        }
```

Comment:
Could simplify this (and make it slightly more efficient) with [`require`][1]:

```suggestion
        auto tmpfile = file.require(
            modulecall ~ " " ~ functioncall,
            fopen(tmpstring.ptr, "w+\0".ptr));
```

[1]: https://dlang.org/library/object/require.html